content/blog/from-cms-to-contextual.md

```diff
@@ -0,0 +1,424 @@
+
+```bash
+export const getStaticProps = async ({ params }) => {
+  const { slug } = params
+  // Ex. `slug` is `hello-world`
+  const variables = { relativePath: `${slug}.md` }
+  // ...
+}
+```
+
+We’ll also want to call `staticRequest` to load our data for our specific page. You’ll also notice that we will return the query & variables from getStaticProps. We’ll be using these values within the TinaCMS frontend.
+
+`import { staticRequest } from 'tinacms'`
+
+So the full query should look like this:
+
+```bash
+export const getStaticProps = async ({ params }) => {
+  const { slug } = params
+  const variables = { relativePath: `${slug}.md` }
+  const data = await staticRequest({
+    query: query,
+    variables: variables,
+  })
+
+  return {
+    props: {
+      data,
+      slug,
+    },
+  }
+}
+```
+
+You may have noticed that our query isn’t there but we are referencing it. This is because we want to be able to reuse the query for our `useTina` hook. At the top of our file, after the imports you can add the query:
+
+```jsx
+const query = `query BlogPostQuery($relativePath: String!) {
+  getPostsDocument(relativePath: $relativePath) {
+    data {
+      title
+      excerpt
+      date
+      coverImage
+      author {
+        name
+        picture
+      }
+      ogImage {
+        url
+      }
+      body
+    }
+  }
+}`
+```
+
+### Adding `useTina` hook
+
+The `useTina` hook is used to make a piece of Tina content editable. It is code-split so in production, this hook will pass through the data value. When you are in edit mode, it registers an editable form in the sidebar. 
+
+### Adding imports
+
+The first thing that needs to be done is import useTina, useEffect and useState. We can also remove a few imports that we will no longer be using. 
+
+```bash
+import {useState, useEffect} from 'react'
+import {useTina} from 'tinacms/dist/edit-state'
+```
+
+### Changing our props and adding UseTina
+
+We are going to update our props from `({ post, morePosts, preview })` to `(props)` . We are going to pass the props into our `useTina` hook. Now that has been updated with the props coming in we can use `useTina` . 
+
+```jsx
+const { data } = useTina({
+    query,
+    variables: props.variables,
+    data: props.data,
+  }) 
+```
+
+As you can see, we are reusing our query from before and passing the variables, and data to `useTina`.  This now means we can power our site using contextual editing. Congratulations your site now has superpowers! 
+
+### Update our elements to use Tina data
+
+Now we have access to our Tina powered data we can go through and update all of our elements to use Tina. You can replace each of the `post.` with `data.getPostsDocument.data.` and then replace the `!post?.slug` with `!props.variables.relativePath` when you are finished your return code should look like:
+
+```jsx
+const router = useRouter()
+  if (!router.isFallback && !props.variables.relativePath) {
```

Comment:
This area was a little confusing because you're changing 3 things in close proximity but you don't get a complete picture of what that looks like. Hard to understand where `useTina` goes relative to the other changes you're making. Maybe provide something like this, or a diff view:

```
export default function Post(props) {

  const { data } = useTina({
    query,
    variables: props.variables,
    data: props.data,
  }) 

  const router = useRouter()
  if (!router.isFallback && !props.variables.relativePath) {
    return <ErrorPage statusCode={404} />
  }
  return (
    <Layout preview={false}>
      <Container>
        <Header />
...
```